README

```diff
@@ -1,3 +1,50 @@
+AT91 U-Boot Repository
+======================
+
+This is a fork of the U-Boot source code for the AT91 family of ARM
+microprocessors.
+
+https://www.microchip.com/design-centers/32-bit-mpus
+
+This repository is intended to make available a version of U-Boot that works on
+all our supported boards at all times. We try to keep this fork as close as
+possible to mainline U-Boot.
+
+The latest version of this repository can be found at:
+
+https://github.com/linux4sam/u-boot-at91
+
+Since U-Boot is closely related to Linux, please read section 3.1 of Microchip
+AN2772:
+
+"Linux® Basics and Solutions for Microprocessors"
+
+http://ww1.microchip.com/downloads/en/Appnotes/Linux-Basics-and-Solutions-for-Microprocessors-Application-Note-DS00002772A.pdf
+
+When porting U-Boot to a new board design which utilises a Microchip AT91 MPU,
+a typical developer workflow might be:
+
+1. With reference to https://www.linux4sam.org/bin/view/Linux4SAM compile
+U-Boot for a supported evaluation board (e.g. an implementation of the same MPU
+that your new design targets).
+
+2. Test the output on the evaluation board.
+
+3. Add support for your new board to your fork of this repository.
+
+4. Repeat steps 1. and 2. against mainline U-Boot:
+
+https://gitlab.denx.de/u-boot/u-boot
+
+5. Compile and test your patch set against mainline U-Boot.
+
+6. Submit your patches to the U-Boot project.
+
```

Comment:
@ehristev : no need to talk about at91 custodian tree here? It might add more confusion: maybe simpler is better, in this case forget my question ;-)

Reply:
@noglitch what do you mean ? is the at91 custodian mentioned anywhere ? I haven't seen it

Reply:
I recommended this particular workflow, because in my case, the mainline u-boot would not boot my board at the time that I tested it (sd card probing bug).  

My assumption was that the https://github.com/linux4sam/u-boot-at91 would be the best tree to start with, since it's generally better tested on at91 SoCs (and then forward-port) - but happy to take whatever guidance you prefer.

Similarly the mainline U-Boot tree and the sam4linux tree are currently mentioned in the PR, but maybe it would be better if https://gitlab.denx.de/u-boot/custodians/u-boot-atmel was mentioned (either as-well, or instead-of the mainline tree).

Perhaps this would be a better workflow:

1. With reference to https://www.linux4sam.org/ compile U-Boot for a supported evaluation board which implements the same MPU that your new design targets.

2. Test your compiled U-Boot on the evaluation board, as far as possible exercising the same functionality that you expect to use on your own board design.

3. Repeat steps 1. and 2. with the latest code from the mainline U-Boot tree at https://gitlab.denx.de/u-boot/u-boot

4. Add support for your new board to a private branch on your copy of the mainline U-Boot source code.

5. Compile and test your patch set against mainline U-Boot.

6. Submit your patches for feedback and integration into the U-Boot project.